app/jobs/concerns/maintenance_tasks/task_job_concern.rb

```diff
@@ -9,6 +9,8 @@ module TaskJobConcern
     include JobIteration::Iteration
 
     included do
+      self.job_iteration_max_job_runtime = JobIteration.max_job_runtime || 5.minutes
```

Comment:
This will still result in us reading the config at the time the concern is included, which may be before the host application sets the value. I think it would make sense to make it dynamic by using a method instead.

```diff
-    included do
+    included do |base|
```

```suggestion
      class << base
        def job_iteration_max_job_runtime = JobIteration.max_job_runtime || 5.minutes
      end
```

although maybe that could be defined in `class_methods` instead... 🤔 

---

From there, I think it would make sense to deprecate, and eventually remove

```ruby
# lib/maintenance_tasks/engine.rb
config.after_initialize do
  JobIteration.max_job_runtime ||= 5.minutes
```

because _really_ this gem shouldn't be setting a global config affecting other jobs (acknowledging the per-job option didn't exist at the time that was added).

We could also make that `5.minutes` configurable as `MaintenanceTasks.max_job_runtime`, if we wanted, and just have `5.minutes` as the default, but that's not a big deal. 🤷